pd-client/client.go

```diff
@@ -125,20 +124,21 @@ func NewClient(pdAddrs []string) (Client, error) {
 }
 
 func (c *client) initClusterID() error {
+	ctx, cancel := context.WithCancel(c.ctx)
+	defer cancel()
 	for i := 0; i < maxInitClusterRetries; i++ {
 		for _, u := range c.urls {
-			// TODO: Use gRPC instead.
-			client, err := apiutil.NewClient(u, pdTimeout)
+			cc, err := c.getOrCreateGRPCConn(u)
 			if err != nil {
 				log.Errorf("[pd] failed to get cluster id: %v", err)
 				continue
 			}
-			clusterID, err := client.GetClusterID()
+			members, err := pdpb.NewPDClient(cc).GetMembers(ctx, &pdpb.GetMembersRequest{})
 			if err != nil {
 				log.Errorf("[pd] failed to get cluster id: %v", err)
 				continue
 			}
-			c.clusterID = clusterID
+			c.clusterID = members.GetHeader().GetClusterId()
```

Comment:
clusterID does not require the cluster has leader. `updateLeader()` will check it later.